In [1]:
import numpy as np # linear algebra 
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
# stats models: regression fitting via formulas
import statsmodels.formula.api as smf
# stats models: regression fitting via matrices of regression design
import statsmodels.api as sm

In [2]:
df = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/Electricity.csv')
df

cost        q        pl      sl      pk      sk      pf      sf
0     0.2130      8.0   6869.47  0.3291  64.945  0.4197  18.000  0.2512
1     3.0427    869.0   8372.96  0.1030  68.227  0.2913  21.067  0.6057
2     9.4059   1412.0   7960.90  0.0891  40.692  0.1567  41.530  0.7542
3     0.7606     65.0   8971.89  0.2802  41.243  0.1282  28.539  0.5916
4     2.2587    295.0   8218.40  0.1772  71.940  0.1623  39.200  0.6606
..       ...      ...       ...     ...     ...     ...     ...     ...
153   6.8293    946.6  10642.16  0.0883  43.600  0.1914  51.463  0.7203
154   3.7605    377.0   7432.24  0.2117  74.120  0.2274  33.436  0.5609
155   3.9822    391.0   5826.04  0.1926  78.288  0.0924  44.633  0.7151
156  30.1880   5317.0   9586.63  0.0845  78.008  0.2009  41.840  0.7147
157  67.8562  15220.0   6986.09  0.1451  74.025  0.2596  25.662  0.5945

[158 rows x 8 columns]

In [3]:
# специфицируем модель через формулу
output_eq1 = smf.ols(formula='np.log(cost)~np.log(q)', data=df).fit()
# Коэфициенты модели с округление
output_eq1.params.round(3)

Intercept   -3.841
np.log(q)    0.836
dtype: float64

$$
log(cost) = -3.841+0.836*log(q)
$$

При увеличении общего выпуска электроэнергии на 1% общие издержки увеличиваются на 0.836%

In [4]:
# специфицируем модель через формулу
output_eq1 = smf.ols(formula='np.log(cost)~np.log(q)+np.log(pl)+np.log(pf)+np.log(pk)', data=df).fit()
# Коэфициенты модели с округление
output_eq1.params.round(3)


Intercept    -7.472
np.log(q)     0.838
np.log(pl)    0.044
np.log(pf)    0.713
np.log(pk)    0.188
dtype: float64

$$
log(cost) = -7.472+0.838*log(q)+ 0.044*log(pl)+0.713* log(pf)+0.188*log(pk)
$$

1. При увеличении общего выпуска электроэнергии на 1% общие издержки увеличиваются на 0.838%
2. При увеличении уровня зарплаты на 1% общие издержки увеличиваются на 0.044%
3. При увеличении цены на топливо на 1% общие издержки увеличиваются на 0.713% 
4. При увеличении цены привлечения капитала на 1% общие издержки увеличиваются на 0.188%